vulkaninfo/vulkaninfo.cpp

```diff
@@ -210,14 +210,15 @@ void DumpPresentableSurfaces(Printer &p, AppInstance &inst, const std::vector<st
             auto exists = surface_list.end();
             for (auto it = surface_list.begin(); it != surface_list.end(); it++) {
                 // use custom comparator to check if the surface has the same values
-                if (it->gpu == gpu.get() && *it->surface == *surface.get()) {
+                if (it->gpu == gpu.get() && *(it->surface) == *(surface.get())) {
                     exists = it;
+                    break;
                 }
             }
             if (exists != surface_list.end()) {
-                exists->surface_types.push_back(surface.get()->surface_extension.name);
+                exists->surface_types.insert(surface.get()->surface_extension.name);
             } else {
-                surface_list.push_back({surface.get(), {surface.get()->surface_extension.name}, gpu.get()});
+                surface_list.push_back({surface.get(), gpu.get(), {surface.get()->surface_extension.name}});
```

Comment:
Is this arg reordering a bugfix separate from fixing the name duplication?  If so, prob merits mention in the PR desc.

Reply:
The SurfaceTypeGroup had arguments in "pointer, vector, pointer" order. It seemed more logical to group the pointers together, which necessitates changing the order of the arguments in the `push_back`.

Reply:
Fair enough.  SurfaceTypeGroup can take either order without modification?

Reply:
I should of specified that SurfaceTypeGroup is a POD-like struct, and mistakenly said arguments when I meant member variables. The order does matter, so the `surface_list.push_back(..)` needed updating.

Reply:
my bad, I misread the diff in SurfaceTypeGroup definition.